CMakeLists.txt

```diff
@@ -91,6 +74,71 @@ if(ENABLE_MKLCPU_BACKEND
   list(APPEND DOMAINS_LIST "rng")
 endif()
 
+# Define required CXX compilers before project
+if(CMAKE_CXX_COMPILER OR NOT ONEMKL_SYCL_IMPLEMENTATION STREQUAL "dpc++")
+  if(WIN32)
+    string(REPLACE "\\" "/" CMAKE_CXX_COMPILER ${CMAKE_CXX_COMPILER})
+  endif()
+else()
+  if(ENABLE_CUBLAS_BACKEND OR ENABLE_CURAND_BACKEND)
+    set(CMAKE_CXX_COMPILER "clang++")
+  elseif(ENABLE_MKLGPU_BACKEND)
+    set(CMAKE_CXX_COMPILER "dpcpp")
+  else()
+    find_program(DPCPP_PATH dpcpp)
+    if(DPCPP_PATH)
+      message(STATUS "dpcpp was found in PATH, using dpcpp")
+      set(CMAKE_CXX_COMPILER "dpcpp")
+    else()
+      if(WIN32)
+        message(STATUS "dpcpp was not found in PATH, using clang-cl instead")
+        set(CMAKE_CXX_COMPILER "clang-cl")
+      else()
+        message(STATUS "dpcpp was not found in PATH, using clang++ instead")
+        set(CMAKE_CXX_COMPILER "clang++")
+      endif()
+    endif()
+  endif()
+endif()
+
+# Define required C compilers before project
```

Comment:
I need to set `CMAKE_CXX_COMPILER` and `CMAKE_C_COMPILER` since dpc++ is not in my path. However, cmake complains that it cannot find clang, even though I have provided a correct path. Please correct me if I am wrong, but here the `CMAKE_C_COMPILER` set by the user will be overwritten, won't it? I believe a similar check might need to be added here as on line 78 to check if the user has set the `CMAKE_C_COMPILER` variable. 

This does not effect hipSYCL I just wanted to share my experience with this change.

Reply:
Thank you very much for catching this problem! We are working on this.